posix/epoll.c

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
```

Comment:
FYI, this is unnecessary here and elsewhere this pattern shows up; clang-format will not sort across whitespace-separated blocks of includes (i.e. enclave.h will not be sorted into the block of headers below)

Side observation: it looks like even after clang-format is turned on, it will no longer sort headers after this. 

Is there any reason these headers need to be in a specific order anyway? For maintainability, I would like there to be no implicit header dependencies that require special ordering. Kind of the point of allowing clang-format to do this is enforce that, and to reduce the dev load of having to think about it.

Reply:
The following change to **<openenclave/internal/thread.h>** requires **<openenclave/enclave.h>** to be included firsts. This should be taken out but is a separate issue:

```
#ifdef _OE_ENCLAVE_H
OE_EXTERNC_BEGIN
```